drivers/sensor/mcp9600/mcp9600.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2020 SER Consulting, LLC
+ * Copyright (c) 2019 Peter Bigot Consulting, LLC
+ * Copyright (c) 2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <kernel.h>
+#include <drivers/i2c.h>
+#include <init.h>
+#include <sys/byteorder.h>
+#include <sys/__assert.h>
+#include <logging/log.h>
+#include "mcp9600.h"
+
+#define DT_DRV_COMPAT microchip_mcp9600
+
+#define MCP9600_BUS_I2C DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+
+LOG_MODULE_REGISTER(MCP9600, CONFIG_SENSOR_LOG_LEVEL);
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "MCP9600 driver enabled without any devices"
+#endif
+
+static int mcp9600_reg_read(const struct device *dev, uint8_t start,
+			uint8_t *buf, int size)
+{
+	struct mcp9600_data *data = dev->data;
+	const struct mcp9600_config *cfg = dev->config;
+
+	return i2c_burst_read(data->i2c_master, cfg->i2c_addr,
+						start, buf, size);
+}
+
+static int mcp9600_sample_fetch(const struct device *dev,
+			enum sensor_channel chan)
+{
+	struct mcp9600_data *data = dev->data;
+	uint8_t buf[2];
+	int size = 2;
+	int ret;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	ret = mcp9600_reg_read(dev, MCP9600_REG_TEMP_HOT, buf, size);
+	if (ret < 0) {
+		return ret;
+	}
+	data->ttemp = (buf[0] << 8) | buf[1];
+
+	return 0;
+}
+
+static int mcp9600_channel_get(const struct device *dev,
+			enum sensor_channel chan,
+			struct sensor_value *val)
+{
+	struct mcp9600_data *data = dev->data;
+	int32_t temp;
+
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		temp = (int32_t)(int16_t)data->ttemp;
```

Comment:
why the double cast?

Reply:
Peter Bigot pointed that out easy way to get it to be sign extended from an uint16_t to int32_t, thought it was kind of elegant. I'll add a comment

Reply:
That'd be from me: the `(int16_t)` is necessary to convert from the unsigned `uint16_t` because the underlying value is signed.  The outer `(int32_t)` I left in my recommendation because IIRC MISRA doesn't like implicit type conversions.

I'm fine with removing the outer one; absent MISRA I wouldn't have it there myself.